src/Blueprint/Blueprint/Cmdlets/RemoveAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Azure.Commands.Blueprint.Models;
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Properties;
+using ParameterSetNames = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterSetNames;
+using ParameterHelpMessages = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterHelpMessages;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintAssignment", SupportsShouldProcess = true)]
+    public class RemoveAzureRmBlueprintAssignment : BlueprintCmdletBase
+    {
+        #region Parameters
+        [Parameter(ParameterSetName = ParameterSetNames.DeleteBlueprintAssignmentByName, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.SubscriptionId)]
+        [ValidateNotNullOrEmpty]
+        public string SubscriptionId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.DeleteBlueprintAssignmentByName, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintAssignmentName)]
+        [ValidateNotNull]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.DeleteBlueprintAssignmentByObject, Position = 0, Mandatory = true, ValueFromPipeline = true, HelpMessage = ParameterHelpMessages.BlueprintAssignmentObject)]
+        public PSBlueprintAssignment BlueprintAssignmentObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+        #endregion Parameters
+
+
+        #region Cmdlet Overrides
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                switch (ParameterSetName)
+                {
+                    case ParameterSetNames.DeleteBlueprintAssignmentByName:
+                        if (ShouldProcess(Name, string.Format(Resources.DeleteAssignmentShouldProcessString, Name, SubscriptionId)))
+                        {
+                            BlueprintClient.DeleteBlueprintAssignment(SubscriptionId, Name);
+
+                            if (PassThru.IsPresent)
+                            {
+                                WriteObject(true);
+                            }
```

Comment:
I think we should return the PSObject representation of what was deleted or nothing if it didn't exist. The following would still be possible in PowerShell with that change because PS has a concept of truthiness similar to JS:
$deletionResult = Remove-AzBlueprintAssignment -dsfjkhdskfjhdskjfhk
if($deletionResult) {} else {}

Reply:
With passThru we can still write the above if statement like below. Using other cmdlets, and looking at their implementation it seemed like this is a common way of implementing the delete functionality. But, it also makes sense to differentiate the result of the delete operation between if the object to be deleted existed or not. I'll ask for recommendation on this during the PR process.
 $deletionResult = Remove-AzBlueprintAssignment -dsfjkhdskfjhdskjfhk -PassThru
if($deletionResult) {} else {}\

---
In reply to: [257385758](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257385758) [](ancestors = 257385758)

Reply:
Makes sense. My point is that we can just return the PS object (if -PassThru switch is set) representing the resource that was deleted if the actual deletion was done. It gives more info that just true/false.

---
In reply to: [257734556](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257734556) [](ancestors = 257734556,257385758)

Reply:
Agreed.

---
In reply to: [258255855](https://github.com/Azure/azure-powershell/pull/8537#discussion_r258255855) [](ancestors = 258255855,257734556,257385758)